libs/wingsdk/src/target-sim/queue.inflight.ts

```diff
@@ -102,12 +104,27 @@ export class Queue implements IQueueClient, ISimulatorResourceInstance {
             type: TraceType.RESOURCE,
             timestamp: new Date().toISOString(),
           });
-          this.messages.push(...messages);
+          this.pushMessagesBackToQueue(messages);
         });
         processedMessages = true;
       }
     } while (processedMessages);
   }
+
+  public async pushMessagesBackToQueue(messages: any): Promise<void> {
+    setTimeout(() => {
+      this.messages.push(...messages);
+      this.context.addTrace({
+        data: {
+          message: `${messages.length} messages pushed back to queue after timeout.`,
+        },
+        sourcePath: this.context.resourcePath,
+        sourceType: QUEUE_TYPE,
+        type: TraceType.RESOURCE,
+        timestamp: new Date().toISOString(),
+      });
+    }, this.timeout * 1000);
```

Comment:
If a function takes 5 seconds to process a message and fail, and the queue's timeout is set for 10 seconds, should we still wait the full 10 seconds before reprocessing it, or should we only wait for the "remaining" time of 5 seconds, so to speak? (I haven't played with this in AWS SQS so I'm not sure off the top of my head how it works there, but figured I'd ask)

Reply:
Good point, I was also thinking about this 😅
 Usually the behaviour is, that the subscriber will delete the message after successfull processing it. During the timeout period, after a subscriber fetched the message from the queue, it remains in the queue but cannot be requested from other subscribers. [This](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-visibility-timeout.html) explanation is really good. To 100% simulate this behaviour the simulator queue might need a rework. This is not trivial I think.
But yes, my fix would currently have a timeout period of 15s if the subscriber throws an error after 5s and and the timeout is 10s. Is it fine or should we simulate this more precise, what do you think?

Reply:
I agree, it should be more accurate in the future, I will create a follow-up feature issue for this ✅